### PR TITLE
Update navigation design for admin settings

### PR DESCRIPTION
**Pull request by 5h1rU**
## Overview
This PR implements the updated navigation design for the admin setting area.

###### **BEFORE**             
Screen Shot 2021-02-24 at 6 45 09 PM

###### **AFTER**             
https://user-images.githubusercontent.com/989826/109346655-fb573000-783f-11eb-9e69-8f858783f860.mov

## Progress
- [X] Documentation text/screenshots updated (if relevant)
- [x] Approved by a frontend engineer (if touching frontend code)
- [x] Approved by a designer (if it touches the UI)

Closes https://github.com/sourcegraph/sourcegraph/issues/13375

### CODE REVIEW

**Comment by AlicjaSuska** [comment]
Thank you for sharing the video @5h1rU! Looks really good already. Here are some things I've noticed:

- we should animate the arrows (I've changed this in the design as well). Section is closed - arrow points down, section is open - arrow points up.
- Icons color is different than on the design
- There is a double border around the section title. There should be a single border. Also, 2 bottom corners shouldn't be rounded.
Now:

image (4)

Design:
Screenshot 2021-02-25 at 11 35 02

Thank you!

**Comment by 5h1rU** [comment]
Thanks @AlicjaSuska 

> Icons color is different than in the design

This dark/light combination is not available in our CSS variables, I can create it, but this will be the only component with those colors. Also, if you look at the user [implementation](https://user-images.githubusercontent.com/9516420/109014388-feb9b280-76ab-11eb-8375-4f1992326362.png), the colors are different. Is this the intended behavior?

**Comment by umpox** [comment]
I think the icon colour is something which I missed on the user page too. I think that's a good point though @5h1rU, this is different to how we display our icon with text components in the nav bar. Would be good to consider aligning this pattern if we're making this change

File: client/web/src/components/Sidebar.scss [hunk 1]
```diff
@@ -1,4 +1,25 @@
 .sidebar {
     isolation: isolate;
     width: 12rem;
+
+    &__btn {
+        background: var(--color-bg-2);
+        border-color: var(--secondary);
```
Comment (limitedmage) [inline client/web/src/components/Sidebar.scss:7]: Note that this color is not themed and will look the same on dark and light mode.

File: client/branded/src/global-styles/colors.scss [hunk 2]
```diff
@@ -122,6 +123,7 @@ $text-muted: var(--text-muted);
     --color-bg-3: #{$gray-19};
     --color-bg-4: #{$gray-22};
     --color-bg-5: #{$gray-17};
+    --color-bg-6: #{$gray-06};
```
Comment (limitedmage) [inline client/branded/src/global-styles/colors.scss:126]: Can we have the design team add these to our color palette in Figma? I saw the Figma design and this color doesn't have a name. cc @AlicjaSuska

File: client/web/src/site-admin/SiteAdminSidebar.story.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        items: [
+            {
+                label: 'Business',
+                to: '/some/business',
+            },
+        ],
+    },
+]
+
+const { add } = storiesOf('web/site-admin/AdminSidebar', module).addDecorator(story => (
```
Comment (limitedmage) [inline client/web/src/site-admin/SiteAdminSidebar.story.tsx:77]: Can we add a Chromatic decorator `.addParameters({ chromatic: { viewports: [$x] }})` or a div wrapper with `style="width:$x"` to limit the width of this component? It looks too wide in the storybook tests

File: client/web/src/site-admin/SiteAdminSidebar.tsx [hunk 4]
```diff
@@ -24,37 +17,40 @@ export interface SiteAdminSidebarProps {
...
  */
 export const SiteAdminSidebar: React.FunctionComponent<SiteAdminSidebarProps> = ({ className, groups }) => (
-    <div className={className}>
+    <div className={`site-admin-sidebar ${className}`}>
         {groups.map(
             ({ header, items, condition = () => true }, index) =>
-                condition({}) && (
-                    <SidebarGroup key={index}>
-                        {header && <SidebarGroupHeader icon={header.icon} label={header.label} />}
+                condition({}) &&
```
Comment (limitedmage) [inline client/web/src/site-admin/SiteAdminSidebar.tsx:23]:
```suggestion
                condition() &&
```
Is the `{}` needed?

Comment (5h1rU) [reply client/web/src/site-admin/SiteAdminSidebar.tsx:23]: condition() [expect an object as context parameter](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/util/contributions.ts#L3), I'm not sure if modifying the interface will produce side effects.

File: client/web/src/components/Sidebar.scss [hunk 5]
```diff
@@ -1,4 +1,25 @@
...
+    &__btn {
+        background: var(--color-bg-2);
+        border-color: var(--border-color);
+        border-radius: 0;
+        &:first-child {
+            border-radius: $border-radius $border-radius 0 0;
+        }
+        &:last-child {
+            border-radius: 0 0 $border-radius $border-radius;
+        }
```
Comment (felixfbecker) [inline client/web/src/components/Sidebar.scss:14]: **suggestion**: This sidenav is a `list-group`. The `list-group-item-action` class for list group items in Bootstrap handles this border radius logic and is set up to be composable with `<button>`s: https://getbootstrap.com/docs/5.0/components/list-group/#links-and-buttons

File: client/web/src/components/Sidebar.tsx [hunk 6]
```diff
@@ -32,6 +35,38 @@ export const SidebarGroupHeader: React.FunctionComponent<{
...
+    children: JSX.Element
+    icon?: React.ComponentType<{ className?: string }>
+    label?: string
+    openByDefault?: boolean
+}> = ({ children, label, icon: Icon, openByDefault = false }) => {
+    const [isOpen, setOpen] = useState<boolean>(openByDefault)
+    const handleOpen = useCallback(() => setOpen(!isOpen), [isOpen])
+    return (
+        <>
+            <Button
```
Comment (felixfbecker) [inline client/web/src/components/Sidebar.tsx:51]: **issue**: for accessibility, the button that controls a collapsible area needs to have `aria-expanded` and `aria-controls` set.

File: client/web/src/site-admin/SiteAdminSidebar.story.tsx [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+            <div style={{ width: '192px' }}>
+                <SiteAdminSidebar className="site-admin-sidebar" groups={[groups[1]]} />
+            </div>
+        </MemoryRouter>
+    ),
+    {
+        design: {
+            type: 'figma',
+            url:
+                'https://www.figma.com/file/cT23UcGqbdjdV4H2yxweIu/%2311775-Map-the-current-information-architecture-%5BApproved%5D?node-id=0%3A1',
```
Comment (felixfbecker) [inline client/web/src/site-admin/SiteAdminSidebar.story.tsx:97]: This link just shows me the whole information architecture design as opposed to just the sidebar displayed in the story:

image

Could you change this to reference the frame that contains the sidebar specifically?

File: client/web/src/enterprise/site-admin/sidebaritems.ts [hunk 8]
```diff
@@ -58,13 +59,7 @@ export const enterpriseSiteAdminSidebarGroups: SiteAdminSideBarGroups = siteAdmi
...
                 {
                     ...group,
-                    items: [
-                        ...group.items,
-                        {
-                            label: 'License',
-                            to: '/site-admin/license',
-                        },
-                    ],
+                    items: group.items,
```
Comment (felixfbecker) [inline client/web/src/enterprise/site-admin/sidebaritems.ts:62]: This is now just referencing the same items from the original `group` so this whole `if` block can be removed :)

File: client/web/src/site-admin/SiteAdminSidebar.story.tsx [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+            },
+        ],
+    },
+]
+
+const { add } = storiesOf('web/site-admin/AdminSidebar', module).addDecorator(story => (
+    <>
+        <div>{story()}</div>
+        <style>{webStyles}</style>
+    </>
```
Comment (felixfbecker) [inline client/web/src/site-admin/SiteAdminSidebar.story.tsx:81]: **fyi**: you can use the `WebStory` component to include the styles (and router too)

File: client/branded/src/global-styles/colors.scss (hunk 2 above)
Comment (felixfbecker) [reply client/branded/src/global-styles/colors.scss:126]: If this gets added, please also make sure it gets added as a `bg-6` helper class by increasing the end number in `backgrounds.scss`: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/branded/src/global-styles/background.scss#L3:1

File: client/web/src/site-admin/SiteAdminSidebar.story.tsx [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+))
+
+add(
```
[the remark this answers is not in the document]
Comment (5h1rU) [reply client/web/src/site-admin/SiteAdminSidebar.story.tsx:11]: Good to know thanks 👍

**Comment by AlicjaSuska** [comment]
@umpox @5h1rU good points, thank you for mentioning the consistency here. I've made some changes to the design:

- icons have regular body color, there is no need to differentiate them, they are part of the label
- chevrons have the same color as clickable icons in the rest of the app ($link) to indicate that this is the clickable element. I know that the whole row is clickable in the accordion but the chevron indicated that (and I wouldn't like to make icon+label+chevron blue, it doesn't look good)

Please, let me know what you think about this option :)

**Comment by 5h1rU** [comment]
@AlicjaSuska I agree. Take a look at the updated video.

**Review by AlicjaSuska — CHANGES_REQUESTED** [review]
Thank you for the updates @5h1rU :) 

- would it be possible to add a divider below the section title? It would be consistent with the user settings visuals. The divider is present in the designs in Figma but may have been hard to notice in combination with the blue background on [this screen](https://www.figma.com/file/cT23UcGqbdjdV4H2yxweIu/11775-Map-the-current-information-architecture-Approved?node-id=68%3A1).

Current:
Screenshot 2021-03-02 at 21 56 40

With divider:
Screenshot 2021-03-02 at 22 00 22

- chevrons should have the $link color (please, see my previous comment for details). Could you update this? Thank you!

- there is one more thing regarding the behavior after clicking on the menu item. I would like us to scroll up to the beginning of the page when user clicks on the menu item so they see the category from the top. I've recorded the [video](https://www.loom.com/share/738209da3e8549f7a089156dd07a45bd) that shows the extreme case when use clicks on 'Subscriptions' and sees the blank part of the page on the right. Scrolling to the top of the page would prevent situations like this. @5h1rU wdyt about this approach?

**Comment by 5h1rU** [comment]
Yes, it makes sense @AlicjaSuska
I implement a scroll top when the route change.

File: client/web/src/site-admin/sidebaritems.ts [hunk 11]
```diff
@@ -95,13 +109,92 @@ export const otherGroup: SiteAdminSideBarGroup = {
...
             to: '/site-admin/report-bug',
         },
+        {
+            label: 'Instrumentation',
+            to: '/-/debug/grafana',
+        },
+        {
+            label: 'Monitoring',
+            to: '/-/debug',
+        },
```
Comment (umpox) [inline client/web/src/site-admin/sidebaritems.ts:119]: Shouldn't these links be swapped? (i.e. instrumentation navigated to `/~/debug` previously)

**Review by umpox — APPROVED** [review]

**Review by AlicjaSuska — APPROVED** [review]
Thank you for the update @5h1rU ! All looks really good :)

One minor thing (not a blocker, so I'm approving the PR): the page could scroll all the way to the top, now it's right at the beginning of the content what looks a little off to me.

Now:
Screenshot 2021-03-04 at 17 56 10

Imporved:
Screenshot 2021-03-04 at 17 56 16

**Pull request merged**